lib/jsdom/events/EventTarget.js

```diff
@@ -0,0 +1,188 @@
+  while (targetParent) {
+    targetList.push(targetParent);
+    target = targetParent;
+    targetParent = target._parentNode;
+  }
+  targetParent = target.defaultView;
+  if (targetParent) {
+    targetList.push(targetParent);
+  }
+
+  var iterator = backwardIterator(targetList);
+
+  event._eventPhase = event.CAPTURING_PHASE;
+  dispatchPhase(event, iterator);
+
+  iterator = singleIterator(event._target);
+  event._eventPhase = event.AT_TARGET;
+  dispatchPhase(event, iterator);
+
+  if (event._bubbles) {
+    iterator = forwardIterator(targetList);
+    event._eventPhase = event.BUBBLING_PHASE;
+    dispatchPhase(event, iterator);
+  }
+
+  event._currentTarget = null;
+  event._eventPhase = event.NONE;
+
+  return !event._preventDefault;
+};
+
+module.exports.EventTarget = EventTarget;
+module.exports.addAlias = instances.addAlias.bind(instances);
+
+function forwardIterator(list) {
+  var i = 0, len = list.length;
+  return function iterator() {
+    return i < len ? list[i++] : null;
+  };
+}
+
+function backwardIterator(list) {
+  var i = list.length;
+  return function iterator() {
+    return i >= 0 ? list[--i] : null;
+  };
+}
+
+function singleIterator(obj) {
+  var i = 1;
+  return function iterator() {
+    return i-- ? obj : null;
+  };
+}
+
+function dispatchPhase(event, iterator) {
+  var target = iterator();
+
+  while (target && !event._stopPropagation) {
+    if (event._eventPhase === event.CAPTURING_PHASE || event._eventPhase === event.AT_TARGET) {
+      callListeners(event, target, getListeners(target, event._type, true));
+    }
+    if (event._eventPhase === event.AT_TARGET || event._eventPhase === event.BUBBLING_PHASE) {
+      callListeners(event, target, getListeners(target, event._type, false));
+    }
+    target = iterator();
+  }
+}
+
+function callListeners(event, target, listeners) {
+  var currentListener = listeners.length;
+  while (currentListener--) {
+    event._currentTarget = target;
+    try {
+      listeners[currentListener].call(target, event);
+    } catch (e) {
+      target.raise(
+        "error", "Dispatching event '" + event._type + "' failed",
+        { error: e, event: event }
+      );
+    }
+  }
+}
+
+function getListeners(target, type, capturing) {
+  var props = instances.get(target);
+
+  var listeners;
+  if (props.events[type]) {
+    listeners = props.events[type][capturing ? "capturing" : "bubbling"];
+  } else {
+    listeners = [];
+  }
+
+  if (!capturing) {
+    var traditionalHandler = target["on" + type];
+    if (traditionalHandler) {
+      var implementation = (target._ownerDocument ? target._ownerDocument.implementation
+                                                  : target.document.implementation);
+
+      if (implementation._hasFeature("ProcessExternalResources", "script")) {
+        if (listeners.indexOf(traditionalHandler) === -1) {
+          listeners.push(traditionalHandler);
+        }
+      }
+    }
+  }
+  return listeners;
+}
```

Comment:
Add a newline at end of file, here and elsewhere. (There should be an option in your IDE to always do this.)